src/sentry/integrations/slack/service.py

```diff
@@ -254,25 +272,40 @@ def _handle_parent_notification(
             "rule_action_uuid": parent_notification.rule_action_uuid,
         }
 
-        try:
-            client.chat_postMessage(
-                channel=channel_id,
-                thread_ts=parent_notification.message_identifier,
-                text=notification_to_send,
-                blocks=json_blocks,
-            )
-            metrics.incr(SLACK_ACTIVITY_THREAD_SUCCESS_DATADOG_METRIC, sample_rate=1.0)
-        except SlackApiError as e:
-            self._logger.info(
-                "failed to post message to slack",
-                extra={"error": str(e), "blocks": json_blocks, **extra},
-            )
-            metrics.incr(
-                SLACK_ACTIVITY_THREAD_FAILURE_DATADOG_METRIC,
-                sample_rate=1.0,
-                tags={"ok": e.response.get("ok", False), "status": e.response.status_code},
-            )
-            raise
+        with MessagingInteractionEvent(
+            interaction_type=MessagingInteractionType.SEND_ACTIVITY_NOTIFICATION,
+            spec=SlackMessagingSpec(),
+        ).capture() as lifecycle:
+            try:
+                client.chat_postMessage(
+                    channel=channel_id,
+                    thread_ts=parent_notification.message_identifier,
+                    text=notification_to_send,
+                    blocks=json_blocks,
+                )
+                # TODO(iamrajjoshi): Remove this after we validate lifecycle
+                metrics.incr(SLACK_ACTIVITY_THREAD_SUCCESS_DATADOG_METRIC, sample_rate=1.0)
+            except SlackApiError as e:
+                # TODO(iamrajjoshi): Remove this after we validate lifecycle
+                self._logger.info(
+                    "failed to post message to slack",
+                    extra={"error": str(e), "blocks": json_blocks, **extra},
+                )
+                metrics.incr(
+                    SLACK_ACTIVITY_THREAD_FAILURE_DATADOG_METRIC,
+                    sample_rate=1.0,
+                    tags={"ok": e.response.get("ok", False), "status": e.response.status_code},
+                )
+                lifecycle.add_extras({"rule_action_uuid": parent_notification.rule_action_uuid})
+                if (
+                    (reason := unpack_slack_api_error(e))
+                    and reason is not None
+                    and reason in SLACK_SDK_HALT_ERROR_CATEGORIES
+                ):
+                    lifecycle.record_halt(reason.message)
+                else:
+                    lifecycle.record_failure(e)
```

Comment:
is this something we're adding to all slack postMessage? i noticed it's also being used in sending a generic message

Reply:
yeah, b/c postMessage happen a lot so we can have more granularity for when stuff failed. this will also increase visibility for how often messages fail to send due to user errors and we can use these metrics to drive future prioritization.

Reply:
if it's going to be reused maybe a helper function would be a good idea ;)

Reply:
done